src/Smuggler/Anns.hs

```diff
@@ -14,3 +18,32 @@ removeAnnAtLoc line col = Map.filterWithKey (\k _ -> matchKey k)
     matchKey (AnnKey (RealSrcSpan rss) _) =
         not (srcSpanStartLine rss == line && srcSpanStartCol rss == col)
     matchKey _ = True
+
+removeTrailingCommas :: Anns -> Anns
+removeTrailingCommas anns
+  = (Map.fromList . concat)
+      (map (\gAnns ->
```

Comment:
It would be more readable if this lambda function is in `where` clause.
Then you can just write:

```haskell
removeTrailingCommas anns = 
    Map.fromList $ concat $ map foo $ groupBy withinSrcSpan $ Map.toList anns
```

or in this case you can eta reduce:

```haskell
removeTrailingCommas = 
    Map.fromList . concat . map foo . groupBy withinSrcSpan . Map.toList
```

As a bonus we can get rid of all this `()` everywhere 🙂